src/frontend/screens/Settings/components/SteamId.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import useSetting from 'frontend/hooks/useSetting'
+import { configStore } from 'frontend/helpers/electronStores'
+import { TextInput } from '@hyperplay/ui'
+
+const SteamId = () => {
+  const { t } = useTranslation()
+  const [steamId, setSteamId] = useSetting('steamId', '')
+
+  function setInstallPath(path: string) {
+    setSteamId(path)
+    configStore.set('settings.steamId', path)
+  }
+
+  return (
+    <TextInput
+      label={t('setting.steamId', 'Steam ID')}
+      value={steamId?.replaceAll("'", '')}
```

Comment:
Should move the `steamId?.replaceAll("'", '')` should be moved inside the `setInstallPath` to prevent rerender
```react

  function handleSteamIdChange(newSteamId: string) {
    const sanitizedSteamId = newSteamId.replaceAll("'", '');
    setSteamId(sanitizedSteamId);
    configStore.set('settings.steamId', sanitizedSteamId);
  }

  return (
    <TextInput
      label={t('setting.steamId')}
      value={steamId}
      placeholder={t('settings.enterSteamId')}
      onChange={(event) => handleSteamIdChange(event.target.value)}
    />
  );
```

Reply:
this is also just for testing, but yes will move it